src/components/OptionList/OptionList.tsx

```diff
@@ -61,129 +64,109 @@ export interface State {
 
 export type CombinedProps = Props & WithAppProviderProps;
 
-export class OptionList extends React.Component<CombinedProps, State> {
-  state: State = {
-    normalizedOptions: createNormalizedOptions(
-      this.props.options,
-      this.props.sections,
-      this.props.title,
-    ),
-  };
-
-  private id = this.props.id || getUniqueId();
-
-  componentWillReceiveProps({
-    options: nextOptions = [],
-    sections: nextSections = [],
-    id: nextID,
-    title: nextTitle,
-  }: Props) {
-    const {options = [], sections = [], id, title} = this.props;
-
-    if (id !== nextID) {
-      this.id = nextID || this.id;
-    }
-
-    const optionsChanged = !arraysAreEqual<OptionDescriptor>(
-      nextOptions,
-      options,
-    );
-
-    const sectionsChanged = !arraysAreEqual<SectionDescriptor>(
-      nextSections,
-      sections,
-      testSectionsPropEquality,
-    );
-
-    const titleChanged = title !== nextTitle;
-
-    if (optionsChanged || sectionsChanged || titleChanged) {
-      this.setState({
-        normalizedOptions: createNormalizedOptions(
-          nextOptions,
-          nextSections,
-          nextTitle,
-        ),
-      });
-    }
+export default function OptionList({
+  options,
```

Comment:
Do you have a reason for preferring destructing in the function signature instead of in the block? I personally don’t like it because it makes signatures harder to read.

Reply:
Not particularly, I just went with it because it's generally the pattern we've been using for functions / functional components.